### PR TITLE
Fix #3384 : Arranged string resources based on screens

### LINKED ISSUE #3384: Arrange string resources based on screens
**Describe the bug**

Currently string resources are not organised correctly and are scattered in `strings.xml` file.

This should be fixed by following these points:
1. Make sure that each string is getting used in one specific view only. For example `Home` is part of multiple views like `Navigation Drawer`, `Home-Toolbar`. So for this we should have separate strings for both.
2. All strings related to one activity/fragment should be kept together by mentioning the activity name on top of the strings in a comment. Example: `<!-- ProfileResetPinActivity -->`
3. All activities/fragment sections should be in correct alphabetical order based on activity/fragment name. (The actual strings name are not required to be in alphabetical order but the comments should be in alphabetical order). Example:
```
<!-- HomeActivity -->
<string name="random_string_related_to_home">ABC</string>
<!-- ProfileResetPinActivity -->
<string name="random_string_related_to_profile_resent_pin">ABC</string>
<string name="random_string_related_to_profile_resent_pin_2">XYZ</string>
```

Arrange strings file as per below criteria:

**Screen/Toolbar Titles**
Used in manifest label, toolbar title and activity label test.
Pattern: `activity_name_activity_title`

Example: `administrator_controls_activty_title`

**App Content Strings**
Pattern: `activity_name_view_component_type`

`administrator_controls_activity_profile_management_label`
`administrator_controls_activity_account_actions_label`
`administrator_controls_activity_logout_item_text`
`administrator_controls_activity_app_version_item`

**Shared Strings**
Pattern: `view_component_description`

reading_text_size_item_text_small
reading_text_size_item_medium

**Test Strings**
Pattern: `test_file_name_view_component_type`

Dialogs
Pattern: `dialog_name_view_component_type`

Grouping Pattern: 

  `<!-- AdministratorControlActivity -->`
….
….
  `<!-- GeneralAvailabilityUpgradeNoticeDialogFragment -->`
…..
   `<!-- ProgressDatabaseFullDialog -->`

### PR DESCRIPTION
**Pull request by MohitGupta121**
## Explanation
Fix #3384 : Arranged string resources based on screens

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

**Comment by MohitGupta121** [comment]
I open this new PR due to this [comment](https://github.com/oppia/oppia-android/pull/4665#issuecomment-1369709109)
As it is due to my mistake I delete my fork repo as that branch is so many commits ahead.

**Review by MohitGupta121 — COMMENTED** [review]
@BenHenning can you please review this PR as reference to your review on my last PR https://github.com/oppia/oppia-android/pull/4665#pullrequestreview-1177955860

**Comment by BenHenning** [comment]
Thanks @MohitGupta121. Will need to look at this tomorrow.

**Comment by BenHenning** [comment]
@MohitGupta121 really sorry will need to look at this on Monday (had an urgent deadline to address this week, but that will be resolved today).

**Comment by MohitGupta121** [comment]
> @MohitGupta121 really sorry will need to look at this on Monday (had an urgent deadline to address this week, but that will be resolved today).

Yes sure, no problem. I understand that release is in 1st priority.

**Comment by BenHenning** [comment]
Need a bit more time, I plan to take at least a partial pass tomorrow.

**Comment by BenHenning** [comment]
Not stale--this is blocked on me reviewing & I still have it backlogged.

**Comment by MohitGupta121** [comment]
Need review from @BenHenning and resolving merge conflicts.

**Comment by BenHenning** [comment]
Hi @MohitGupta121. Apologies, I couldn't quite get to this today with my other PR reviews. I plan to look at this tomorrow or Wed since I now have a shortcut way to more quickly check it.

**Comment by BenHenning** [comment]
I plan to look at this tomorrow or possibly Thursday, barring any high priority things that come my way. :)

File: app/src/main/res/values/strings.xml
Comment (BenHenning) [inline app/src/main/res/values/strings.xml]: This value has appeared to have regressed. Shouldn't it be: ``Tap to change the voiceover language.``?

File: app/src/main/res/values/strings.xml
Comment (BenHenning) [inline app/src/main/res/values/strings.xml]: Is this string actually being used? It doesn't seem to be and I think it was removed in #4601.

File: app/src/main/res/values/strings.xml
Comment (BenHenning) [inline app/src/main/res/values/strings.xml]: I believe all three of these no longer need the asterisk per #4601. Maybe this was missed during a sync with develop?

File: app/src/main/res/values/strings.xml
Comment (BenHenning) [inline app/src/main/res/values/strings.xml]: This text is out-of-date. It should be: ``Before we add profiles, we need to protect your account by creating a PIN. This gives you the ability to authorize manage profiles on the device.``.

File: app/src/main/res/layout-sw600dp-land/profile_chooser_fragment.xml
Comment (BenHenning) [inline app/src/main/res/layout-sw600dp-land/profile_chooser_fragment.xml:1]: This file was deleted in #4849.

**Review by BenHenning — DISMISSED** [review]
Thanks @MohitGupta121 both for the changes & your patience in me taking quite a while to get back to this.

I created a script to help me verify the correctness of the changes (which I want to pin down first, then I'll focus on naming specifics). I left some comments, but my script also detected the following changes being needed:

```
Discovered incorrect ID renames:
- An occurrence of policy_activity_title was changed to policy_activity_label_title, but expected one of:
  - privacy_policy_activity_title

- An occurrence of recently_played_stories was changed to profile_progress_header_activity_recently_played_stories, but expected one of:
  - recently_played_stories

- An occurrence of reading_text_size was changed to reading_text_size_activity_title, but expected one of:
  - reading_text_size

- An occurrence of recently_played_stories was changed to profile_progress_header_activity_recently_played_stories, but expected one of:
  - recently_played_stories

- An occurrence of reading_text_size was changed to reading_text_size_activity_title, but expected one of:
  - reading_text_size

- An occurrence of recently_played_stories was changed to profile_progress_header_activity_recently_played_stories, but expected one of:
  - recently_played_stories

- An occurrence of privacy_policy_title was changed to privacy_policy_activity_title, but expected one of:
  - policy_activity_label_title

- An occurrence of privacy_policy_title was changed to privacy_policy_activity_title, but expected one of:
  - policy_activity_label_title

- An occurrence of privacy_policy_title was changed to privacy_policy_activity_title, but expected one of:
  - policy_activity_label_title

- An occurrence of privacy_policy_title was changed to privacy_policy_activity_title, but expected one of:
  - policy_activity_label_title

- An occurrence of privacy_policy_title was changed to privacy_policy_activity_title, but expected one of:
  - policy_activity_label_title

- An occurrence of privacy_policy_title was changed to privacy_policy_activity_title, but expected one of:
  - policy_activity_label_title

- An occurrence of privacy_policy_title was changed to privacy_policy_activity_title, but expected one of:
  - policy_activity_label_title

- An occurrence of privacy_policy_title was changed to privacy_policy_activity_title, but expected one of:
  - policy_activity_label_title

- An occurrence of privacy_policy_title was changed to privacy_policy_activity_title, but expected one of:
  - policy_activity_label_title
```

Unfortunately, it won't say where these are but they seem to be straightforward to find. Can you look into this, the other comments, and then sync with the latest develop before sending back to me? I'll take another correctness pass, and then probably review everything except the strings in detail (since that's the largest part).

**Comment by MohitGupta121** [comment]
@BenHenning Thanks you so much for giving time to review this. I will update this PR as per review and update all missing strings as per develop.

File: app/src/main/res/values/strings.xml
Comment (MohitGupta121) [reply app/src/main/res/values/strings.xml]: Done.

File: app/src/main/res/values/strings.xml
Comment (MohitGupta121) [reply app/src/main/res/values/strings.xml]: Done.

File: app/src/main/res/values/strings.xml
Comment (MohitGupta121) [reply app/src/main/res/values/strings.xml]: Done.

File: app/src/main/res/values/strings.xml
Comment (MohitGupta121) [reply app/src/main/res/values/strings.xml]: Done.

File: app/src/main/res/layout-sw600dp-land/profile_chooser_fragment.xml
Comment (MohitGupta121) [reply app/src/main/res/layout-sw600dp-land/profile_chooser_fragment.xml:1]: Done.

**Comment by MohitGupta121** [comment]
> ```
> Discovered incorrect ID renames:
> - An occurrence of policy_activity_title was changed to policy_activity_label_title, but expected one of:
>   - privacy_policy_activity_title

Hey @BenHenning, if possible can you please explain what exactly this "Discovered incorrect ID renames" detected by your script in this https://github.com/oppia/oppia-android/pull/4823#pullrequestreview-1355847791 ? What I exactly need to change. Please reply whenever you free no rush.
Thanks!

**Comment by MohitGupta121** [comment]
@adhiamboperes PTAL, Thanks! As per comment of Ben, he transfers the review of this PR to you:



Screenshot 2023-08-02 at 12 51 04 PM

**Comment by MohitGupta121** [comment]
@adhiamboperes PTAL, Thanks!

File: app/src/main/AndroidManifest.xml
Comment (adhiamboperes) [inline app/src/main/AndroidManifest.xml:49]: The label should be for the activity and not title.

File: app/src/main/AndroidManifest.xml
Comment (adhiamboperes) [inline app/src/main/AndroidManifest.xml:53]:
```suggestion
      android:label="@string/faq_list_activity_title"
```

File: app/src/main/AndroidManifest.xml
Comment (adhiamboperes) [inline app/src/main/AndroidManifest.xml:117]: Please make sure that the activity/toolbar title strings are not confused with the label strings that are used in the options fragment. I think you will notice that you need 2 seperate strings for this.

File: app/src/main/AndroidManifest.xml
Comment (adhiamboperes) [inline app/src/main/AndroidManifest.xml:144]: This does not conform to the expected pattern.

File: app/src/main/res/values/strings.xml
Comment (adhiamboperes) [inline app/src/main/res/values/strings.xml:4]: There is another instance of AdministratorControlActivity grouping on line 25, shouldn't they be together?

File: app/src/main/res/values/strings.xml
Comment (adhiamboperes) [inline app/src/main/res/values/strings.xml:25]: Per issue description, all activities should be in the correcte alphabetical order, so we should have something like **AdminAuthActivity**, followed by **AdministratorControlsActivity** the **AdminPinActivity**. **AddProfileActivity** will come before **AppVersionActivity** and so on.

File: app/src/main/res/values/strings.xml
Comment (adhiamboperes) [inline app/src/main/res/values/strings.xml:42]: This does not need to be it's own category as it is still part of `administrator_controls_*`

File: app/src/main/res/values/strings.xml
Comment (adhiamboperes) [inline app/src/main/res/values/strings.xml:75]: Th
```suggestion
  <!-- AudioDialogFragment -->
```

File: app/src/main/res/values/strings.xml
Comment (adhiamboperes) [inline app/src/main/res/values/strings.xml:84]:
```suggestion
  <!-- CellularAudioDialogFragment -->
```

File: app/src/main/res/values/strings.xml
Comment (adhiamboperes) [inline app/src/main/res/values/strings.xml:118]: These are in the same category, and `stop_exploration_dialog_fragment_stop_exploration_dialog_title` can move to the bottom of this category.

File: app/src/main/res/values/strings.xml
Comment (adhiamboperes) [reply app/src/main/res/values/strings.xml]: Looks like a typo, should be `Before we add profiles, we need to protect your account by creating a PIN. This gives you the ability to manage profiles on the device.`

I think that the text we are removing is `authorize downloads and`

File: app/src/main/res/values/strings.xml
Comment (adhiamboperes) [inline app/src/main/res/values/strings.xml:175]:
```suggestion
  <string name="hints_and_solution_summary_hide_solution">Hide solution</string>
  <string name="hints_and_solution_summary_show_solution">Show solution</string>
```

File: app/src/main/res/values/strings.xml
Comment (adhiamboperes) [inline app/src/main/res/values/strings.xml:168]: This sting is duplicated in ln174 below

File: app/src/main/res/values/strings.xml
Comment (adhiamboperes) [inline app/src/main/res/values/strings.xml:176]:
```suggestion
  <string name="reveal_solution_dialog_fragment_message">This will show the solution. Are you sure?</string>
```

File: app/src/main/res/values/strings.xml
Comment (adhiamboperes) [inline app/src/main/res/values/strings.xml:177]:
```suggestion
  <string name="reveal_solution_dialog_fragment_text">Reveal</string>
```

File: app/src/main/res/values/strings.xml
Comment (adhiamboperes) [inline app/src/main/res/values/strings.xml:180]:
```suggestion
  <string name="input_interaction_activity_text_input_default_content_description">Enter text.</string>
```

File: app/src/main/res/values/strings.xml
Comment (adhiamboperes) [inline app/src/main/res/values/strings.xml:181]:
```suggestion
  <string name="input_interaction_activity_numeric_input_hint">Enter a number.</string>
```

File: app/src/main/res/values/strings.xml
Comment (adhiamboperes) [inline app/src/main/res/values/strings.xml:242]: I saw that these are used in the MathExpressionInteractionsViewModel, and not in an activity so they should probably revert to their original naming, and this category heading should change from `MathExpressionActivity` to  `MathExpressionInteractionsViewModel`

File: app/src/main/res/values/strings.xml
Comment (adhiamboperes) [inline app/src/main/res/values/strings.xml:289]: PinPasswordActivity comes before ProfileChooserFragment and ProgressDatabaseFullDialogActivity in alphabetical order

File: app/src/main/res/values/strings.xml
Comment (adhiamboperes) [inline app/src/main/res/values/strings.xml:357]: I think these two strings have been switched.

File: app/src/main/res/values/strings.xml
Comment (adhiamboperes) [inline app/src/main/res/values/strings.xml:373]: RevisionCardActivity comes after RecentlyPlayedActivity

**Review by adhiamboperes — CHANGES_REQUESTED** [review]
Hey @MohitGupta121 , thanks for organizing this file!

In addition to my inline comments, 

- [ ] Let's resolve the activity title rename issue once it's been resolved in the discussion.
- [ ] You do no need to make changes to the string files in the other languages, translatewiki will do the updates.
- [ ] Please resolve the merge conflicts.

**Comment by MohitGupta121** [comment]
> * [ ]  You do no need to make changes to the string files in the other languages, translatewiki will do the updates.

@adhiamboperes Thank you so much for taking the time to review this PR. 
Basically, if I change a string's name in the string.xml file then I need to also change the name of that string to other language files otherwise that string is unresolved in other languages. Right? So how do we ignore that?

**Comment by adhiamboperes** [comment]
> @adhiamboperes Thank you so much for taking the time to review this PR. Basically, if I change a string's name in the string.xml file then I need to also change the name of that string to other language files otherwise that string is unresolved in other languages. Right? So how do we ignore that?

@MohitGupta121, we resolved in team meeting to change the names in all translations so that we do not loose all our translations, but you need to file an issue to follow up on translatewiki to remove the "fuzzy" attribute that will likely be added.

**Pull request closed without merging**
